src/parsing_routines_matrixnetworks.jl

```diff
@@ -122,3 +122,130 @@ function loadrxnetwork(ft::MatrixNetwork,
+
+- The complex incidence matrix, is number of complexes by number of reactions with
+  Bᵢⱼ = -1, if the i'th complex is the substrate of the j'th reaction,
+         1, if the i'th complex is the product of the j'th reaction,
+         0, otherwise
+"""
+
+struct ComplexMatrixNetwork{S ,T ,U ,V ,W,X}
+    """The symbolic expressions for each reaction rate."""
+    rateexprs::S
+
+    """stoichmat[i,j] = is the stoichiometric coefficient in the j'th reaction for the i'th species"""
+    stoichmat::T
+
+    """incidencemat[i,j] is the incidence matrix with incidencemat[i,j] = -1, if the i'th complex
+    is the substrate of the j'th reaction, 1, if the i'th complex is the product
+    of the j'th reaction, 0, otherwise"""
+    incidencemat::U  # all elements are integer always in incidence matrix
+
+    """species in the network """
+    species::V
+
+    """ Parameters """
+    params::W
+
+    """independent variable, time """
+    t::X
+end
+ComplexMatrixNetwork(rateexprs,stoichmat,incidencemat; species=Any[],params=Any[],t=nothing) =
+           ComplexMatrixNetwork(rateexprs,stoichmat,incidencemat,species,params,t)
+
+
+
+# for Dense matrices version
+function loadrxnetwork(cmn::ComplexMatrixNetwork{S,T,U,V,W,X}) where {S <: AbstractVector,
+                T <: Matrix, U <: Matrix{Int}, V <: AbstractVector,
+                                        W <: AbstractVector,X <: Any}
+   numspecs, numcomp = size(cmn.stoichmat)
+   @assert all(>=(0),cmn.stoichmat)
+   @assert numcomp == size(cmn.incidencemat, 1)
+   @assert all(∈([-1,0,1]),cmn.incidencemat)
+   numrxs = size(cmn.incidencemat, 2)
+
+   t = (cmn.t === nothing) ? (@variables t)[1] : cmn.t
+   species = isempty(cmn.species) ? [funcsym(:S,t,i) for i = 1:numspecs] : cmn.species
+
+   rn = Vector{Reaction}(undef,numrxs)
+   sc_ind = argmin(cmn.incidencemat, dims=1)  # cartesian indices of substrate complexes
+   pc_ind = argmax(cmn.incidencemat, dims=1)  # cartesian indicies of products complexes
+
+   for i ∈ 1:numrxs
+
+       # substrate index for i'th reaction in species(rn)
+       ss_ind = findall(!iszero, @view cmn.stoichmat[:,sc_ind[i][1]])
+
+       # products index for i'th reaction in species(rn)
+       ps_ind = findall(!iszero, @view cmn.stoichmat[:,pc_ind[i][1]])
+
+       if isempty(ss_ind) && !isempty(ps_ind)
+           rn[i] = Reaction(cmn.rateexprs[i], nothing, species[ps_ind],
+                       nothing, cmn.stoichmat[ps_ind,pc_ind[i][1]])
+
+       elseif !isempty(ss_ind) &&  isempty(ps_ind)
+           rn[i] = Reaction(cmn.rateexprs[i], species[ss_ind], nothing,
+                       cmn.stoichmat[ss_ind,sc_ind[i][1]], nothing)
+       else
+           rn[i] = Reaction(cmn.rateexprs[i], species[ss_ind],species[ps_ind],
+                       cmn.stoichmat[ss_ind,sc_ind[i][1]],
+                       cmn.stoichmat[ps_ind,pc_ind[i][1]])
+       end
+   end
+   @named rs = ReactionSystem(rn,t,species, cmn.params)
```

Comment:
This probably isn't a good idea as it will mean any system loaded in this way is named `rs`. I'd suggest changing to:
```julia
rs = ReactionSystem(rn, t, species, cmn.params; name=gensym(:ReactionSystem))
```
and changing below too.

Reply:
yes..makes sense :+1: